framework/src/modules/dpos/transaction_assets/unlock_transaction_asset.ts

```diff
@@ -105,25 +112,36 @@ export class UnlockTransactionAsset extends BaseAsset<UnlockTransactionAssetCont
 				throw new Error('Corresponding unlocking object not found.');
 			}
 
-			const waitingPeriod = getWaitingPeriod(
-				sender.address,
-				delegate.address,
-				store.chain.lastBlockHeaders[0].height,
-				unlock,
-			);
+			const currentHeight = store.chain.lastBlockHeaders[0].height + 1;
+			const minWaitingHeight = getMinWaitingHeight(sender.address, delegate.address, unlock);
 
-			if (waitingPeriod > 0) {
+			if (minWaitingHeight > currentHeight) {
```

Comment:
`minWaitingHeight == currentHeight` should be ok here?

Reply:
maybe this is question to @MaximeGagnebin

Reply:
so, `minWaitingHeight == currentHeight` should be ok, indeed. 
Isn't that what the logic does? like if `>` then throw error?

Reply:
yes indeed 😅